File: cases/051_quad/matlab/main.m
% 051_quad - 数值积分
% 使用quad函数进行数值积分

%% 定义被积函数
f = @(x) x.^2;

%% 数值积分
I = quad(f, 0, 1);  % ∫₀¹ x² dx = 1/3

%% 显示结果
fprintf('∫₀¹ x² dx = %f\n', I);
fprintf('精确值: %f\n', 1/3);

File: cases/051_quad/python/main.py
#!/usr/bin/env python3
"""
051_quad - 数值积分
使用scipy.integrate进行数值积分
"""

import numpy as np
from scipy.integrate import quad

def f(x):
    return x**2

# 计算积分 ∫₀¹ x² dx
I, error = quad(f, 0, 1)

print(f"∫₀¹ x² dx = {I:.6f}")
print(f"精确值: {1/3:.6f}")
print(f"误差: {abs(I - 1/3):.10f}")
